### PR TITLE
fix: remove defaultProps

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog
fix: remove defaultProps

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix: remove defaultProps |
| 🇨🇳 Chinese | 删除 defaultProps |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by li-jia-nan** [comment]
偶然间发现我发现 `spread` + `defaultProps` 有时候会把属性透传到 dom 节点上，所以需要传给 dom 的属性，还是尽量手动 `pick` 出来，少用 `spread` 或者 `omit` 方法

File: components/input/Password.tsx [hunk 1]
```diff
@@ -3,13 +3,13 @@ import EyeOutlined from '@ant-design/icons/EyeOutlined';
...
 import omit from 'rc-util/lib/omit';
 import * as React from 'react';
-
 import { useState } from 'react';
 import type { ConfigConsumerProps } from '../config-provider';
 import { ConfigConsumer } from '../config-provider';
 import type { InputProps, InputRef } from './Input';
 import Input from './Input';
 
+const defaultIconRender = (v: boolean) => (v ? <EyeOutlined /> : <EyeInvisibleOutlined />);
```
Comment (MadCcc) [inline components/input/Password.tsx:12]:
```suggestion
const defaultIconRender = (visible: boolean) => (visible ? <EyeOutlined /> : <EyeInvisibleOutlined />);
```

File: components/input/Password.tsx [hunk 2]
```diff
@@ -30,24 +30,23 @@ const Password = React.forwardRef<InputRef, PasswordProps>((props, ref) => {
     if (disabled) {
       return;
     }
-
-    setVisible(!visible);
+    setVisible(v => !v);
```
Comment (MadCcc) [inline components/input/Password.tsx:33]: 尽量不要用一个字母命名变量

File: components/input/Password.tsx [hunk 3]
```diff
@@ -30,24 +30,23 @@ const Password = React.forwardRef<InputRef, PasswordProps>((props, ref) => {
...
-    const iconTrigger = ActionMap[action!] || '';
+    const { action = 'click', iconRender = defaultIconRender } = props;
+    const iconTrigger = ActionMap[action] || '';
     const icon = iconRender(visible);
     const iconProps = {
       [iconTrigger]: onVisibleChange,
       className: `${prefixCls}-icon`,
       key: 'passwordIcon',
-      onMouseDown: (e: MouseEvent) => {
+      onMouseDown(e: MouseEvent) {
```
Comment (MadCcc) [inline components/input/Password.tsx:44]: 我更偏向箭头函数。。这个感觉没必要改

**Review by MadCcc — APPROVED** [review]

**Pull request merged**